web/profiles/joinup/joinup.behat.inc

```diff
@@ -105,28 +105,27 @@ class JoinupSubContext extends DrupalSubContextBase implements DrupalSubContextI
    */
   public function submitRegistrationForm(TableNode $table, $location) {
     $this->visitPath($location);
+    $username = '';
 
     // Fill in fields.
     foreach ($table->getRowsHash() as $field => $value) {
       $this->getSession()->getPage()->fillField($field, $value);
+      if ($field == 'Username') {
+        $username = $value;
+      }
     };
     // Submit form.
     $this->getSession()->getPage()->pressButton("Create new account");
 
     // Get the last registered user.
-    $storage = \Drupal::entityManager()->getStorage('user');
-    $query = $storage->getQuery();
-    $result = $query->sort('uid', 'DESC')->range(0, 1)->execute();
-    $uid = array_pop($result);
+    $results = \Drupal::entityManager()->getStorage('user')
+        ->loadByProperties(['name' => $username]);
```

Comment:
What if the username is empty? This logic doesn't seem valid to me. Isn't the username a required field? I would throw an exception if the username is missing.